File: Lab2/app/src/main/java/com/example/lab2/MainActivity.java
package com.example.lab2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    public static String IP = "ip";
    public static String NICK = "nick";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText serverEditText = findViewById(R.id.serverEditText);
        EditText nicknameEditText = findViewById(R.id.nicknameEditText);

        Button startButton = findViewById(R.id.startButton);

        startButton.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, SimpleChatActivity.class);
            intent.putExtra(IP, serverEditText.getText().toString());
            intent.putExtra(NICK, nicknameEditText.getText().toString());
            startActivity(intent);
        });
    }
}

File: Lab2/app/src/main/java/com/example/lab2/MainActivity.kt
package com.example.lab2

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    companion object {
        val IP = "ip"
        val NICK = "nick"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val serverEditText = findViewById<EditText>(R.id.serverEditText)
        val nicknameEditText = findViewById<EditText>(R.id.nicknameEditText)

        val startButton = findViewById<Button>(R.id.startButton)

        startButton.setOnClickListener {
            val server = serverEditText.text.toString()
            val nickname = nicknameEditText.text.toString()

            if (server != "" && nickname != "") {
                val intent = Intent(this@MainActivity, SimpleChatActivity::class.java)
                intent.putExtra(IP, server)
                intent.putExtra(NICK, nickname)
                startActivity(intent)
            } else {
                Toast.makeText(
                    this,
                    "Please provide correct server and nickname",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
    }
}